src/udev/udev-node.c

```diff
@@ -690,7 +690,7 @@ int udev_node_apply_permissions(
                         } else if (streq(name, "smack")) {
                                 smack = true;
 
-                                q = mac_smack_apply_fd(node_fd, SMACK_ATTR_ACCESS, label);
+                                q = mac_smack_apply(devnode, SMACK_ATTR_ACCESS, label);
```

Comment:
Not familiar with SMACK, but how about to use `fd_reopen()`?

Reply:
Sure, but why not add `O_RDWR` in the `open` call in line 300 - the end state would be the same.

Reply:
Not sure about SELinux, but, at least `fchmod_and_chown()` does not require `O_RDWR`, no? So, I'd like to open files in rw on demand.

Reply:
Let me give a shot in my internal environment.

Reply:
After some more digging, I think commit a7fdc6c was the root cause as it opened the fd using the `O_PATH` option. This will cause `fsetxattr()` to fail with a bad file descriptor error according to the man page for open.

I'm trying to see if I can do a `fd_reopen` with the right mode so the labelling will go through.